capstone/fitness/src/main/java/com/google/sps/fit/Exercise.java

```diff
@@ -0,0 +1,211 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.fit;
+
+import java.io.Serializable;
+import java.util.*;
+
+/* Representation of exercise with specific quantitative data. */
+public class Exercise implements Serializable {
+  
+  public enum SetType {
+    DISTANCE,
+    DURATION_INC,
+    DURATION_DEC,
+    REPS,
+    WEIGHT;
+    
+    /**
+     * Certain measurements for exercises want to decrease in number whereas others don't. This addresses that by
+     * returning a "greater than" comparison for two numbers depending on if more of the stat is better.
+     *
+     * @param src starting float
+     * @param comp float to compare to
+     * @return if the starting float is "greater than" its comparison.
+     */
+    public boolean greaterThan(float src, float comp) {
+      if(this.name().contains("_DEC")) {
+        return src < comp;
+      }
+      return src > comp;
+    }
+  }
+
+  /* Exercise name. */
+  private String name;
+
+  /* Sets per exercise. */
+  private int sets;
+
+  /* Set type mappings to values. */
+  private HashMap<SetType, float[]> setValues;
+
```

Comment:
Why are you mapping the set types to values. Wouldn't it make more sense for the Exercise class to have 1 set type (distance, or reps, or etc) and an array of sessions with the data recorded from each one?

Reply:
I am mapping set types to values to make scaling to other exercises other than timed running easier (e.g. bench press has a reps and weight associated with it). For the MVP, only one set type is needed, and this framework still supports it.

To compare the latest sessions to the Progress Model as a whole, there needed to be a layer of separation with some common ground. The Exercise class acts as that common ground such that the Session data can be used to create Exercises that directly compare to the GoalStep's, in the Progress Model, Exercises.